src/lwc/geFormRenderer/geFormRenderer.js

```diff
@@ -2080,18 +2171,39 @@ export default class GeFormRenderer extends LightningElement{
     }
 
     updateFormStateWithSuccessfulPurchaseCall(responseBody) {
-        this.updateFormState({
-            [apiNameFor(PAYMENT_ELEVATE_ID)]: responseBody.id,
-            [apiNameFor(PAYMENT_STATUS)]: responseBody.status,
-            [apiNameFor(PAYMENT_CARD_NETWORK)]: responseBody.cardData.brand,
-            [apiNameFor(PAYMENT_LAST_4)]: responseBody.cardData.last4,
-            [apiNameFor(PAYMENT_EXPIRATION_MONTH)]: responseBody.cardData.expirationMonth,
-            [apiNameFor(PAYMENT_EXPIRATION_YEAR)]: responseBody.cardData.expirationYear,
-            [apiNameFor(PAYMENT_DECLINED_REASON)]: '',
-            [apiNameFor(PAYMENT_GATEWAY_ID)]: responseBody.gatewayId,
-            [apiNameFor(PAYMENT_TRANSACTION_ID)]: responseBody.gatewayTransactionId,
-            [apiNameFor(PAYMENT_AUTHORIZED_AT)]: responseBody.authorizedAt,
-        });
+        if (this.isCreditCardTransaction()) {
+            this.updateFormState({
+                [apiNameFor(PAYMENT_ELEVATE_ID)]: responseBody.id,
+                [apiNameFor(PAYMENT_STATUS)]: responseBody.status,
+                [apiNameFor(PAYMENT_CARD_NETWORK)]: responseBody.cardData.brand,
+                [apiNameFor(PAYMENT_LAST_4)]: responseBody.cardData.last4,
+                [apiNameFor(PAYMENT_EXPIRATION_MONTH)]: responseBody.cardData.expirationMonth,
+                [apiNameFor(PAYMENT_EXPIRATION_YEAR)]: responseBody.cardData.expirationYear,
+                [apiNameFor(PAYMENT_DECLINED_REASON)]: '',
+                [apiNameFor(PAYMENT_GATEWAY_ID)]: responseBody.gatewayId,
+                [apiNameFor(PAYMENT_TRANSACTION_ID)]: responseBody.gatewayTransactionId,
+                [apiNameFor(PAYMENT_AUTHORIZED_AT)]: responseBody.authorizedAt,
+            });
+        } else {
+            this.updateFormState({
+                [apiNameFor(PAYMENT_ELEVATE_ID)]: responseBody.id,
+                [apiNameFor(PAYMENT_STATUS)]: responseBody.status,
+                [apiNameFor(PAYMENT_DECLINED_REASON)]: '',
+                [apiNameFor(PAYMENT_GATEWAY_ID)]: responseBody.gatewayId,
+                [apiNameFor(PAYMENT_TRANSACTION_ID)]: responseBody.gatewayTransactionId,
+                [apiNameFor(PAYMENT_AUTHORIZED_AT)]: responseBody.authorizedAt,
+                [apiNameFor(PAYMENT_ACH_LAST_4)]: responseBody.achData.last4,
+                [apiNameFor(PAYMENT_ACH_CODE)]: responseBody.achData.achCode,
+                [apiNameFor(PAYMENT_METHOD)]: responseBody.paymentType,
+                [apiNameFor(PAYMENT_ACH_CONSENT)]: responseBody.achData.consentMessage,
+                [apiNameFor(PAYMENT_TYPE)]: responseBody.paymentType,
+                [apiNameFor(PAYMENT_ELEVATE_ORIGINAL_PAYMENT_ID)]: responseBody.originalTransactionId
+            });
```

Comment:
Would it be possible to create a base object here and then only add the few properties that are conditional after checking isCreditCardTransaction to avoid the duplicate lines?